.github/workflows/test.yml

```
@@ -134,35 +140,49 @@ jobs:
           name: playwright-report-${{ matrix.os }}
           path: playwright-report
 
-      - name: Upload updated snapshots to artifact
-        if: steps.check-whether-to-update-snapshots.outputs.shouldUpdateSnapshots == 'true'
+      - name: Collect patch for snapshots
+        if: needs.config.outputs.shouldUpdateSnapshots == 'true'
+        run: |
+          git add --intent-to-add tests/  # git diff に表示されるようにする
+          git diff tests/  # ロギング用
+          git diff --binary tests/ > patch-${{ matrix.os }}.diff
+
+      - name: Upload patch to artifact
+        if: needs.config.outputs.shouldUpdateSnapshots == 'true'
         uses: actions/upload-artifact@v4
         with:
           name: updated-snapshots-${{ matrix.os }}
-          path: tests/e2e/browser/*-snapshots/*
+          path: patch-${{ matrix.os }}.diff
 
   commit-snapshots:
     runs-on: ubuntu-latest
     permissions:
       contents: write
-    needs:
-      - e2e-test
+    needs: [config, e2e-test]
+    if: needs.config.outputs.shouldUpdateSnapshots == 'true'
     steps:
       - uses: actions/checkout@v3
 
       - name: Download artifacts
         uses: actions/download-artifact@v4
         with:
           pattern: updated-snapshots-*
-          path: tests/e2e/browser
+          path: patches
           merge-multiple: true
 
       - name: Commit updated snapshots
         run: |
-          git config --global user.name "github-actions[bot]"
-          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+          # パッチを適用
+          for patch in patches/*.diff; do
+            git apply $patch
```

```suggestion
            git apply --allow-empty $patch
```
これだとどうなります？空のdiffが渡されて落ちてそうだったので

---

あれ、付けてたと思ってました！
ありがとうございます！！